src/feature/relay/selftest.c

```diff
@@ -242,7 +255,9 @@ router_do_reachability_checks(int test_or, int test_dir)
 
   if (router_should_check_reachability(test_or, test_dir)) {
     if (test_or && (!orport_reachable || !circuit_enough_testing_circs())) {
-      router_do_orport_reachability_checks(me, orport_reachable);
+      /* We'll separate IPv4 and IPv6 reachability detection in #34067. */
```

Comment:
I'm guessing that this comment means we need separate "orport_reachable" variables for an ipv6 and ipv4 address

Reply:
We'll need separate variables, and potentially separate flags that make up each variable. I clarified the comment in e887887cd32f9e78e3b6e9c47661eaf750b4c0fb.

See also https://trac.torproject.org/projects/tor/ticket/33222 , particularly:
> 4.2.3. Separate IPv4 and IPv6 Reachability Flags
> 
> Testing relays (and bridges) will record reachability separately for IPv4
> and IPv6 ORPorts, based on the ORPort that the test circuit was received on.
> 
> Here is a reliable way to do reachability self-tests for each ORPort:
> 
> 1. Check for create cells on inbound ORPort connections 
> 
> Check for a cell on any IPv4 and any IPv6 ORPort. (We can't know which listener(s) correspond to the advertised ORPorts, particularly when using port forwarding.) Make sure the cell was received on an inbound OR connection.
> 
> 2. Check for created cells from testing circuits on outbound OR connections 
> 
> Check for a returned created cell on our IPv4 and IPv6 self-test circuits. Make sure those circuits were on outbound OR connections.
> 
> By combining these tests, we confirm that we can:
> * reach our own ORPorts with testing circuits
> * send and receive cells via inbound OR connections to our own ORPorts
> * send and receive cells via outbound OR connections to other relays' ORPorts 
> 
> This isn't a perfect test, there are a few false positives...